node.go

```diff
@@ -125,26 +128,39 @@ func (n *funcNode) value(g *Container) (reflect.Value, error) {
 	cv := reflect.ValueOf(n.constructor)
 
 	values := cv.Call(args)
-	v := values[0]
 
-	// If the function has two return values, the second one is an error.
-	var err error
-	if len(values) > 1 {
-		err, _ = values[1].Interface().(error)
+	// cache constructed values in the node
+	for _, node := range n.nodes {
+		for _, v := range values {
+			if node.objType == v.Type() {
+				n.cached = true
+				n.cachedValue = append(n.cachedValue, v)
+			}
+		}
 	}
 
-	n.cached = true
-	n.cachedValue = v
-	return v, err
+	// last value must be an error
+	err, _ := values[len(values)-1].Interface().(error)
```

Comment:
Fixing the comment. Its not "must be an error". Its only checking if last value is an error and returning it.